src/ui/public/filter_bar/lib/multi_filter.js

```diff
@@ -0,0 +1,93 @@
+import _ from 'lodash';
+
+/**
+ * Create single filter per data field. Fields with multiple filters are merged into a bool query.
+ * Filters without meta.key can not be merged because field is unknown.
+ *
+ * @param {array} filters to merge.
+ * @returns {array} merged filters
+ */
+export function mergeFilters(filters) {
+  // Group filters by key
+  const filterGroups = new Map();
+  const filtersWithNoKey = [];
+  filters.forEach((filter) => {
+    const key = _.get(filter, 'meta.key');
+    if (!key) {
+      filtersWithNoKey.push(filter);
+      return;
+    }
+
+    if (!filterGroups.has(key)) {
+      filterGroups.set(key, [filter]);
+    } else {
+      filterGroups.set(key, filterGroups.get(key).concat(filter));
+    }
+  });
+
+  const filtersWithKey = [];
+  filterGroups.forEach((filterGroup, key) => {
+    if (filterGroup.length === 1) {
+      // Key with only one filter
+      filtersWithKey.push(filterGroup[0]);
+    } else {
+      // Key with multiple filters - combine into OR query
+      const values = [];
+      const shouldFilters = [];
+      const mustNotFilters = [];
+      filterGroup.forEach((filter) => {
+        const value = _.get(filter, 'meta.value', '');
+        if (_.get(filter, 'meta.negate', false)) {
+          values.push('!' + value);
+          mustNotFilters.push(cleanFilter(filter));
+        } else {
+          values.push(value);
+          shouldFilters.push(cleanFilter(filter));
+        }
+      });
+      filtersWithKey.push({
+        bool: {
+          should: shouldFilters,
+          must_not: mustNotFilters
+        },
+        meta: {
+          alias: `${key}: ${values.join()}`
+        }
+      });
+    }
+  });
+
+  return filtersWithKey.concat(filtersWithNoKey);
+}
+
+function cleanFilter(filter) {
+  if (_.has(filter, 'query')) {
+    return filter.query;
+  }
+
+  delete filter.meta;
+  return filter;
+}
+
+const CTRL_KEY = 17;
+const MACOS_COMMAND_KEY = 93;
```

Comment:
This didn't work on OSX for me. It looks like it's browser dependent and you'll need to check for multiple codes: https://stackoverflow.com/questions/3902635/how-does-one-capture-a-macs-command-key-via-javascript